File: examples/typescript/validate-invoice.ts
import { Peppol } from "@getpeppr/sdk";

const peppol = new Peppol({ apiKey: "sk_sandbox_..." });

// Validate locally before sending — catches errors instantly, no API call needed
const result = peppol.validate({
  number: "INV-2026-042",
  to: {
    name: "Globex NV",
    peppolId: "0208:BE0987654321",
    street: "Rue de la Loi 200",
    city: "Brussels",
    postalCode: "1000",
    country: "BE",
  },
  lines: [
    { description: "Consulting Q1", quantity: 40, unitPrice: 125, vatRate: 21 },
  ],
});

if (!result.valid) {
  for (const error of result.errors) {
    console.error(`[${error.field}] ${error.message}`);

    if (error.suggestion) {
      console.log(`  Tip: ${error.suggestion}`);
    }
  }
}

// Also returns warnings for non-blocking suggestions
for (const warning of result.warnings) {
  console.warn(`[${warning.field}] ${warning.message}`);
}

File: examples/python/validate_invoice.py
"""Validate an invoice via the getpeppr API before sending."""

import requests

BASE_URL = "https://api.getpeppr.dev"
API_KEY = "sk_sandbox_abc123..."

response = requests.post(
    f"{BASE_URL}/v1/validate",
    headers={
        "Authorization": f"Bearer {API_KEY}",
        "Content-Type": "application/json",
    },
    json={
        "number": "INV-2026-042",
        "to": {
            "name": "Globex NV",
            "peppolId": "0208:BE0987654321",
            "street": "Rue de la Loi 200",
            "city": "Brussels",
            "postalCode": "1000",
            "country": "BE",
        },
        "lines": [
            {
                "description": "Consulting Q1",
                "quantity": 40,
                "unitPrice": 125,
                "vatRate": 21,
            },
        ],
    },
    timeout=30,
)

response.raise_for_status()
result = response.json()

# /validate returns errors as strings and checks required fields only.
# Use /validate/server for gateway-side UBL and business-rule checks.
if not result["valid"]:
    for error in result["errors"]:
        print(f"Validation error: {error}")
else:
    print("Required fields are present; this does not prove Peppol sendability.")
